lib/jsonapi/view.ex

```diff
@@ -205,8 +210,21 @@ defmodule JSONAPI.View do
         end)
       end
 
-      @impl View
-      def fields, do: raise("Need to implement fields/0")
+      cond do
+        !@polymorphic_resource ->
+          @impl View
+          def fields, do: raise("Need to implement fields/0")
+
+          @impl View
+          def polymorphic_fields(_data), do: nil
```

Comment:
The type of this callback does not include `nil` (same for the other new callback).

Reply:
If in practice this function just gets ignored whenever `@polymorphic_resource` is `false`, may as well return `[]` (which fits its type spec) instead of `nil`.

Reply:
An alternative design might be explicitly allow `nil` and avoid the need for the `@polymorphic_type` boolean by instead defining the `resource_type/1` function to be `polymorphic_type(data)` if it isn't `nil` and otherwise `type()`. Having the `@polymorphic_type` boolean is probably the better option, though.